app/ts/background/background-startup.ts

```diff
@@ -147,7 +148,11 @@ async function newBlockAttemptCallback(blockheader: EthereumBlockHeader, ethereu
 	await sendPopupMessageToOpenWindows({ method: 'popup_new_block_arrived', data: { rpcConnectionStatus } })
 	if (isNewBlock) {
 		const settings = await getSettings()
```

Comment:
```ts
  await setRpcConnectionStatus(rpcConnectionStatus)
	await updateExtensionBadge()
	await sendPopupMessageToOpenWindows({ method: 'popup_new_block_arrived', data: { rpcConnectionStatus } })
	if (isNewBlock) {
		const settings = await getSettings()
```
If these throw its fine? Just the promises inside the try/catch is intended to to be a non-blocking error?

Reply:
You are right its not fine. I just didn't expect them to throw, but so I ignored them. But we should also catch errors from them as well if they happen to throw. I fixed that